File: Xamarin-Mac/Xamarin-Mac/MyFrameLoadDelegate.cs
using System;
using Foundation;
using ObjCRuntime;
using WebKit;
using JavaScriptCore;

namespace XamarinMac {
    partial class MyFrameLoadDelegate : WebFrameLoadDelegate {
        JSContext jsContext { get; set; }
        MyNativeBridge nativeBridge { get; set; }

        [Export("webView:didCreateJavaScriptContext:forFrame:")]
        public void DidCreateJavascriptContext(WebView view, JSContext context, WebFrame frame) {
            jsContext = context;
            jsContext.ExceptionHandler = (JSContext contextException, JSValue exception) => {
                Console.WriteLine("JavaScript Exception: {0}", exception);
            };

            nativeBridge = new MyNativeBridge(jsContext);
            jsContext[(NSString)"nativeBridge"] = JSValue.From(nativeBridge, jsContext);
        }
    }
}

File: Swift/SwiftDesktop/SwiftDesktop/MyFrameLoadDelegate.swift
//
//  MyFrameLoadDelegate.swift
//  SwiftDesktop
//
//  Copyright (c) 2015 DesktopJavaScript. All rights reserved.
//

import Cocoa
import WebKit
import JavaScriptCore

class MyFrameLoadDelegate: NSObject {
    
    override func webView(sender: WebView!, didCreateJavaScriptContext context: JSContext!, forFrame: WebFrame!) {
        let myNativeBridge = MyNativeBridge(newContext: context)
        context.setObject(myNativeBridge, forKeyedSubscript: "nativeBridge")
    }
}
